great_expectations/self_check/util.py

```diff
@@ -913,27 +912,14 @@ def build_sa_validator_with_data(  # noqa: C901, PLR0912, PLR0913, PLR0915
     if context is None:
         context = build_in_memory_runtime_context()
 
-    my_data_connector: ConfiguredAssetSqlDataConnector = ConfiguredAssetSqlDataConnector(
-        name="my_sql_data_connector",
-        datasource_name="my_test_datasource",
-        execution_engine=execution_engine,
-        assets={
-            "my_asset": {
-                "table_name": "animals_table",
-            },
-        },
-    )
-
     if batch_definition is None:
-        batch_definition = (
-            my_data_connector.get_batch_definition_list_from_batch_request(
-                batch_request=BatchRequest(
-                    datasource_name="my_test_datasource",
-                    data_connector_name="my_sql_data_connector",
-                    data_asset_name="my_asset",
-                )
-            )
-        )[0]
+        # maintain legacy behavior - standup a dummy LegacyBatchDefinition
+        batch_definition = LegacyBatchDefinition(
+            datasource_name="my_test_datasource",
+            data_connector_name="my_sql_data_connector",
+            data_asset_name="my_asset",
+            batch_identifiers=IDDict(),
+        )
```

Comment:
Previously, this function used the `ConfiguredAssetSqlDataConnector` to build a `LegacyBatchDefinition`. After running through a bunch of code, it appears that the batch definition never has `batch_identifiers`, and all the other fields are dummy fields not corresponding to real data. The solution is to just stand up the `LegacyBatchDefinition`, and tests appear to be satisfied.